### ISSUE TITLE
id field of type BSON Binary is converted to String.

### ISSUE THREAD

**Issue by vbuzzano** [balderdashy/sails-mongo#229]
Hello,

I face the problem that when loading a record that has Binary type for it's id, the Binary object is converted to String. Wich is a problem.

**Comment by vbuzzano** [comment]
I think you should change the code rewriteIds 

...
  if(typeof model._id === 'object')
        model.id = model._id.toString()
...

by

...
  if(typeof model._id === 'object'** && !(model._id instanceof Binary)**)
    model.id = model._id.toString();
...

**Comment by listepo** [comment]
+1

**Comment by vkopitsa** [comment]
+1

**Comment by sailsbot** [comment]
Thanks for posting, @vbuzzano.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
